app/lib/services/import.dart

```diff
@@ -382,17 +385,55 @@ class ImportService {
     return null;
   }
 
+// ! start
   Future<NoteData?> importPdf(Uint8List bytes, NoteData document,
       {Offset? position,
       bool createAreas = false,
       bool advanced = true}) async {
     try {
+      final dialog = showLoadingDialog(context);
       final firstPos = position ?? Offset.zero;
       final elements = <Uint8List>[];
       final localizations = AppLocalizations.of(context);
+      // Define the compression value
+      int level = 9;
+      // Define the lot size
+      const batchSize = 100;
+      List<Uint8List> batch = [];
+      // Create a counter for decoded images
+      int decodedImagesCount = 0;
+      // Get the total number of pages
+      int totalPages = await Printing.raster(bytes).length;
```

Comment:
I would save Printing.raster in a variable and use this to improve performance.

Reply:
I can't put it in a variable, this leads me to the StateError error (Bad state: Stream has already been listened to.), because I can't listen to the stream multiple times, I would have to convert the stream to a list but I think this would be worse than calling it back several times